File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        k %= len(nums)
        right = nums[len(nums)-k:]
        nums[len(nums)-k:] = []
        nums[:0] = right

File: 0189-rotate-array/0189-rotate-array.java
class Solution {
  public void rotate(int[] nums, int k) {
    k %= nums.length;
    int[] temp = Arrays.copyOfRange(nums, nums.length - k, nums.length);
    for (int i = nums.length - 1; i - k >= 0; i--) {
      nums[i] = nums[i - k];
    }
    for (int i = 0; i < temp.length; i++) {
      nums[i] = temp[i];
    }
  }
}